### PR TITLE
Add Datepicker to GoalForm

### PR DESCRIPTION
**Pull request by ConnorBoulais**
This PR adds a datepicker to the GoalForm component and also adds some 'alerts' that notify the user if they try to create a goal without filling out the required information.

### CODE REVIEW

File: src/Components/GoalForm.js [hunk 1]
```diff
@@ -65,21 +79,61 @@ function GoalForm() {
...
+          <div className="datePicker">
+            <KeyboardDatePicker
+              wrapperClassName="datePicker"
+              disableToolbar
+              variant="inline"
+              format="yyyy-MM-DD"
+              margin="normal"
+              id="date-picker-inline"
+              label="Due Date"
+              value={dueDate}
```
Comment (samiur98) [inline src/Components/GoalForm.js:103]: Good job incorporating Material UI.

File: src/Components/GoalsScreen.css [hunk 2]
```diff
@@ -287,4 +287,25 @@ button:active {
   border-radius: 1em;
   padding: 2em;
   border-width: 0.1em;
+}
+
+.datePicker {
+  display: inline;
+}
+
+.datePicker button{
```
Comment (samiur98) [inline src/Components/GoalsScreen.css:296]: Good job with the CSS.

**Review by samiur98 — APPROVED** [review]
Looks good to be merged in.

File: package-lock.json [hunk 3]
```diff
@@ -25332,6 +25332,19 @@
       "resolved": "https://registry.npmjs.org/@csstools/normalize.css/-/normalize.css-10.1.0.tgz",
       "integrity": "sha512-ij4wRiunFfaJxjB0BdrYHIH8FxBJpOwNPhhAcunlmPdXudL1WQV1qoP9un6JsEBAgQH+7UXyyjh0g7jTxXK6tg=="
     },
+    "@date-io/core": {
```
Comment (tddumont) [inline package-lock.json:25335]: Excellent work here!

**Review by tddumont — APPROVED** [review]
Great job figuring out the DatePicker!

**Pull request merged**